service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java

```diff
@@ -0,0 +1,155 @@
+    String[] expectedMessage;
+
+    @Before
+    public void initialize() {
+        expectedErrorMessage = "kapua-locator-service-error-messages";
+        kapuaLocatorErrorCodes = new KapuaLocatorErrorCodes[]{KapuaLocatorErrorCodes.SERVICE_UNAVAILABLE, KapuaLocatorErrorCodes.SERVICE_PROVIDER_INVALID,
+                KapuaLocatorErrorCodes.FACTORY_UNAVAILABLE, KapuaLocatorErrorCodes.FACTORY_PROVIDER_INVALID, KapuaLocatorErrorCodes.COMPONENT_UNAVAILABLE,
+                KapuaLocatorErrorCodes.COMPONENT_PROVIDER_INVALID, KapuaLocatorErrorCodes.INVALID_CONFIGURATION};
+        argument1 = "argument1";
+        argument2 = "argument2";
+        argument3 = "argument3";
+        throwables = new Throwable[]{new Throwable(), null};
+        expectedMessageNullArguments = new String[]{"Service unavailable {0}", "{0} is not a valid service provider for {1}", "Factory unavailable {0}",
+                "{0} is not a valid factory provider for {1}", "Error: ", "Error: ", "Error: "};
+        expectedMessage = new String[]{"Service unavailable " + argument1, argument1 + " is not a valid service provider for " + argument2, "Factory unavailable " + argument1,
+                argument1 + " is not a valid factory provider for " + argument2, "Error: " + argument1 + ", " + argument2 + ", " + argument3, "Error: " + argument1 + ", " + argument2 + ", " + argument3,
+                "Error: " + argument1 + ", " + argument2 + ", " + argument3};
+    }
+
+    @Test
+    public void kapuaLocatorExceptionKapuaLocatorErrorCodesParameterTest() {
+        for (int i = 0; i < kapuaLocatorErrorCodes.length; i++) {
+            KapuaLocatorException kapuaLocatorException = new KapuaLocatorException(kapuaLocatorErrorCodes[i]);
+            assertEquals("Expected and actual values should be the same.", kapuaLocatorErrorCodes[i], kapuaLocatorException.getCode());
+            assertNull("Null expected.", kapuaLocatorException.getCause());
+            assertEquals("Expected and actual values should be the same.", expectedMessageNullArguments[i], kapuaLocatorException.getMessage());
+            assertEquals("Expected and actual values should be the same.", expectedErrorMessage, kapuaLocatorException.getKapuaErrorMessagesBundle());
+        }
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void kapuaLocatorExceptionKapuaLocatorNullErrorCodesParameterTest() {
+        KapuaLocatorException kapuaLocatorException = new KapuaLocatorException(null);
+        assertNull("Null expected.", kapuaLocatorException.getCode());
+        assertNull("Null expected.", kapuaLocatorException.getCause());
+        assertEquals("Expected and actual values should be the same.", expectedErrorMessage, kapuaLocatorException.getKapuaErrorMessagesBundle());
+        kapuaLocatorException.getMessage();
+    }
+
+    @Test
+    public void kapuaLocatorExceptionKapuaLocatorErrorCodesObjectParametersTest() {
+        for (int i = 0; i < kapuaLocatorErrorCodes.length; i++) {
```

Comment:
Please use enhanced for loop.

Reply:
I used index i to get values from different arrays in the same position